### ISSUE TITLE
Date and DateTime saves wrong date

### ISSUE THREAD

**Issue by gerchomat** [Gernott/mask#234]
If i have a Date or DateTime field in a content element it saves the wrong date at saving.
Every time i save the content element the datepicker generates a date with the wrong timezone.

You can reproduce it if you add a Date or DateTime field to your content element and save the content element. 

In my project i have the timezoe "Europe/Berlin". 

Mask: 4.1.0
TYPO3 9.5.7 (also tested with 9.5.5)

**Comment by gerchomat** [comment]
It only occurs with an "mask generated" Date or DateTime field, not with mask generated starttime or endtime field.

The difference between these two fields (date & starttime e.g.) are the genrated tca and the generated sql.

**Comment by gerchomat** [comment]
The generated tca for a date-field contains a "dbType=date" and the generated sql is "date".
I think this is wrong cause TYPO3 works with integer sql-columns

**Comment by gerchomat** [comment]
So if i change the sql for "date" field to:

`<input type="hidden" name="tx_mask_tools_maskmask[storage][sql][{type}][--index--]" value="int(11) DEFAULT '0' NOT NULL"/>`

And remove the "dbType" from the template it works without any problem.

So i don't know about any downwards compatibility, but this solves the problem.

**Comment by Gernott** [comment]
I cannot reproduce this. Works fin in my setup.
dbType is correct, see: https://docs.typo3.org/m/typo3/reference-tca/master/en-us/ColumnsConfig/Type/Input.html#dbtype
It looks like a problem in your system environment?

**Comment by nitori** [comment]
I think it's a general problem with how TYPO3 stores dates using dbType.

Note: I have both Europe/Berlin on my browser and on the server.

When a new datetime value is input into a datetime field, something like "14:00 02-01-2020", the value that is being transmitted to the server is actually "2020-01-02T14:00:00Z", the Z indicating it's a UTC datetime.

It basically pretends that the input is a UTC date, no matter what timezone the users Browser is in. This "conversion" is done by the JavaScript FormEngine.

The database itself is stored with the (naive) datetime "2020-01-02 14:00:00".

When opening the content element again and inspecting the hidden field that contains the current value of the datetime field, It might contain a value like "2020-01-02T15:00:00+01:00" instead. 

The stored DB value was apparently read as UTC, and converted to the local server timezone and then rendered into the field. JavaScript though reads it and converts it internally to UTC again, to display the correct "14:00" in the form field.

So in my opinions: A lot of nasty stuff is being done when using dbType, and there is a case where using that field type fails the user: Extbase.

When you map a datetime field to an extbase Model, and annotate that field with "@var \DateTime", the value that was stored in the database, ("14:00") is interpreted as UTC, and typo3 maps it to a DateTime object in *local* time, meaning the DateTime contains a value like "2020-01-02T15:00:00+01:00".

Using f:format.date etc. in a ViewHelper will therefor output the incorrect 15:00 time, instead of the intended 14:00 time.

---

To summarize: The date that is input into the field, is a local time (from perspective of the editor), but it's interpreted as UTC by TYPO3, which might cause issues in some cases, like the Extbase one.

In General I stopped using dbType and use int(11) for the SQL fields instead. That always seemed to be more stable in my experience so far.
